### PR TITLE
CC-KBC | Provide compilation features for different TEE attesters

### LINKED ISSUE #212: CC-KBC | Provide compilation features for different TEE attesters
Now CC-KBC compiles all the TEE attesters it supports by default, which is redundant. We need to set different features to specify which TEE attester the compiled CC-KBC contains.

cc @fitzthum @Xynnn007

### PR DESCRIPTION
**Pull request by jialez0**
Resolve: confidential-containers/guest-components#212

### CODE REVIEW

**Comment by jialez0** [comment]
cc @fitzthum

File: .github/workflows/cc_kbc.yml [hunk 1]
```diff
@@ -47,6 +47,7 @@ jobs:
       - name: Build and install with cc_kbc feature
         run: |
           make KBC=cc_kbc && make install
+          make KBC=cc_kbc_tdx && make install
```
Comment (dubek) [inline .github/workflows/cc_kbc.yml:50]: The new `make` invocation doesn't fit the step's `name`.
Also below on the clippy step we don't use the `cc_kbc_tdx` feature.

Suggestsion: Consider adding a matrix dimension above (line 22):

```yaml
      matrix:
        rust:
          - stable
        kbc:
          - cc_kbc
          - cc_kbc_tdx
```

and then use `${{ matrix.kbc }}` in each of the steps to choose the current one.

**Review by Xynnn007 — COMMENTED** [review]
Great. Some same work is involved in another PR https://github.com/confidential-containers/attestation-agent/pull/136/files#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542R52 but it's good to me to rebase after this PR

A little suggestion: probably we can also add rust fmt check in the cc-kbc CI for different attesters in the way that @dubek suggested?

**Comment by jialez0** [comment]
> A little suggestion: probably we can also add rust fmt check in the cc-kbc CI for different attesters in the way that @dubek suggested?

@Xynnn007 In `basic.yml`, we use `cargo fmt --all -- --check`, this checked all features of AA, so in `cc_kbc.yml`, we don't need to check the format again.

File: .github/workflows/cc_kbc.yml [hunk 2]
```diff
@@ -46,11 +49,11 @@ jobs:
 
       - name: Build and install with cc_kbc feature
```
Comment (dubek) [inline .github/workflows/cc_kbc.yml:50]: maybe change this to:

```
name: Build and install with ${{ matrix.kbc }} feature
```

(assuming that you can use these interpolations in the `name` field; not sure)

**Review by dubek — APPROVED** [review]
Looks good. Small comment below:

**Review by fitzthum — APPROVED** [review]
LGTM. Ideally we will build for all platforms in production. I will rebase my SNP attester to use.

**Pull request merged**